src/parser.cpp

```diff
@@ -498,11 +501,11 @@ void Parser::ParserImpl::loadUnit(const UnitsPtr &units, const XmlNodePtr &node)
                 err->setKind(Error::Kind::UNITS);
                 err->setRule(SpecificationRule::UNIT_MULTIPLIER);
                 mParser->addError(err);
-            } else {
-                multiplier = std::stod(attribute->getValue());
             }
         } else if (attribute->isType("offset")) {
-            if (mParser->catchDoubleConversionError(attribute->getValue())) {
+            if (canConvertToDouble(attribute->getValue())) {
+                offset = std::stod(attribute->getValue());
+            } else {
```

Comment:
I am saying that `canConvertToDouble()` tells us whether we can convert to a double by actually trying to do the conversion. So, it's kind of a waste of time to do another conversion when `canConvertToDouble()` returns `true`. Instead, we might as well pass `offset` so that `canConvertToDouble()` can save the result of the conversion it tries to do, and we then can use that information rather than re-computing it.